torch/testing/_internal/common_methods_invocations.py

```diff
@@ -145,6 +145,104 @@ def _getattr_qual(obj, name, default=_NOTHING):
+#     3) A "sample_inputs_func", a function that can generate valid inputs to
+#          the operator. Generating valid inputs is important for "black box"
+#          testing, like PyTorch's autograd tests, which can infer if an
+#          operator is behaving correctly just by observing how changes in
+#          to an operator's inputs affect its gradient computation.
+#
+# Most OpInfos are collected in the "op_db" sequence, although some specialized
+#   tests have their own OpInfo-based datastructures.
+#
+# OpInfos are intended to replace custom test generators, like "method_tests()"
+#   (see below) and the "tensor_op_tests" in test_torch.py. These generators
+#   proved hard to read and maintain, and having multiple operator lists
+#   was frequently confusing, with engineers not understanding what each
+#   list actually did. OpInfos hopes to address these issues and extend
+#   automated test coverage by consolidating these lists into one structured and
+#   documented list.
+#
+# Most tests that use every OpInfo are in test_ops.py. More correctly,
+#   test_ops.py contains "test templates" which are instantiated by
+#   PyTorch's "device generic test framework". See the developer Wiki article
+#   above for more information on this test instantiation process.
+#
+# In addition to the OpInfo base class, there are subclasses of OpInfos, like
+#   UnaryUfuncInfo, that extend the base class's supported metadata for
+#   a group of operator with a common structure. For example, the
+#   UnaryUfuncInfos have a "ref" (short for "reference") attribute that defines
+#   a NumPy or SciPy reference for them. These subclasses may have additional
+#   tests in their corresponding files. UnaryUfuncInfos, for example, have
+#   specialized tests in test_unary_ufuncs.py. In the future BinaryUfuncInfos
+#   and ReductionInfos will be created and specialized tests added for them,
+#   too.
+#
+# When an OpInfo is added tests for it are automatically generated based on
+#   its metadata and the templated tests defined in PyTorch's test suite
+#   that use the @ops decorator. Again, for more on this process see the
+#   linked developer Wiki article. Knowing what, exactly, is tested when
+#   adding an OpInfo requires reading the corresponding test files, but
+#   test_ops.py, where the majority of tests live, can be thought of as
+#   validating the following (as of this writing):
+#
+#     1) That an operator's function, method, and inplace variants (if they
+#          exist) perform the same computation.
+#     2) That an operator's aliases, if any, perform the same computation.
+#     3) That an operator implements its gradient and second-order gradient
+#          correctly.
+#     4) That an operators works when traced or scripted by the jit.
+#     5) That operators supporting out= do so correctly.
+#     6) That operators specify metadata, like which dtypes they support,
+#          properly.
```

Comment:
It would make sense to mention here that some (or all?) of these tests run only on the first element of  tuple returned by `sample_inputs_func`